src/hotspot/cpu/aarch64/register_aarch64.hpp

```diff
@@ -144,21 +146,23 @@ class FloatRegisterImpl: public AbstractRegisterImpl {
   };
 
   // construction
-  inline friend FloatRegister as_FloatRegister(int encoding);
+  inline friend constexpr FloatRegister as_FloatRegister(int encoding);
 
-  VMReg as_VMReg();
+  VMReg as_VMReg() const;
 
   // derived registers, offsets, and addresses
-  FloatRegister successor() const                          { return as_FloatRegister((encoding() + 1) % 32); }
+  FloatRegister successor() const { return as_FloatRegister((encoding() + 1) % 32); }
```

Comment:
Should this `32` be replaced by `number_of_registers`. Furthermore, an `&` would save some instructions here, a `static_assert` to ensure `number_of_registers` is a power of 2, too.

Reply:
I think I'll leave that optimization out, but I will change `32` to `number_of_registers`.

`successor()` is used only once in release code, it's not worth optimizing.
Also, that `% 32` is a kludge I'd like to get rid of, but this patch isn't supposed to affect anything but UB.